agents/supervisor/supervisor.go

```diff
@@ -52,9 +53,12 @@ type Supervisor struct {
 	qanRequests   chan agentpb.QANCollectRequest
 	l             *logrus.Entry
 
-	m              sync.Mutex
+	rw             sync.RWMutex
 	agentProcesses map[string]*agentProcessInfo
 	builtinAgents  map[string]*builtinAgentInfo
+
+	amx           sync.Mutex
+	agentStatuses map[string]inventorypb.AgentStatus
```

Comment:
❗️The big problem is that we never delete elements from that map. We should do it.

Reply:
Fixed